kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -179,11 +179,20 @@ simplifyClaimRule =
     worker :: ClaimPattern -> LogicT simplifier ClaimPattern
     worker claimPattern = do
         let lhs = ClaimPattern.left claimPattern
-        simplified <- simplify lhs
-        let substitution = Pattern.substitution simplified
-            lhs' = simplified { Pattern.substitution = mempty }
+            rhs = ClaimPattern.right claimPattern
+        simplifiedLhs <- simplify lhs
+        simplifiedRhs <- OrPattern.filterOr <$> traverse simplify rhs
```

Comment:
@ana-pantilie Aren't we already simplify the right-hand side of the claim somewhere else?

Reply:
Yes, in the `simplify` transition of the strategy: https://github.com/kframework/kore/blob/c28200738e0a5e8c2b9aaf6a6046b996f68082e3/kore/src/Kore/Strategies/Goal.hs#L838